### ISSUE TITLE
Ability for user provided products to be displayed in flow run UI

### ISSUE THREAD

**Issue by tjgalvin** [PrefectHQ/prefect#7598]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

The Orion web UI provides a nice and responsive way of monitoring and visualizing the progression of a flow run. At the moment the information displayed are mostly logs, start/end times of the flow run, the radar figure etc. 

### Describe the proposed behavior

It would be useful to provide a facility within the prefect flow to upload additional user provided data products that would also be displayed on the web UI. For instance, one of my regular pipelines produces 5 png files. If there was a way that these could be uploaded onto the orion server so that they could be displayed in the web UI of that flow run it would be immediately valuable. 

In short, introduce a very basic facility to record diagnostic information (text, pngs) of the data processed by a flow run recorded along side the information already presented by the orion web UI. This would allow a single location to immediately know the state of the flow run and the state of the data processed by said flow run.

**Comment by peytonrunyan** [comment]
Thanks for submitting the enhancement issue! I think it's probably unlikely that we'll support this exact use-case in the near future. 

Off the top of my head though, you could log the text information to have it immediately available. If you have a task for logging the text and task for logging the URLs of the generated images (say you upload them to an s3 bucket), then you just click into the task, and there you will have the specific logs that you're looking for, and the URLs of your images..
